dds/DCPS/XTypes/DynamicData.cpp

```diff
@@ -1367,7 +1464,8 @@ bool DynamicData::read_values(SequenceType& value, TypeKind elem_tk)
   case TK_STRING8:
   case TK_STRING16:
     {
-      return strm_ >> value;
+      // TODO: Help.
```

Comment:
The `DdsDynamicData.idl` file defines various sequence types as part of the Dynamic Data Interface.  The serializer doesn't understand these types.  Thus, we need to decide if we typedef them to the ones it does understand, switch to the new ones, etc.

Reply:
No.  But they will be deserialized.

To be specific, if I uncomment these lines, the compiler complains that the operation is not defined for the DDS::*Seq type.  It is defined for the sequence types we are used to dealing with.  Thus, it seems I can either typedef DDS::*Seq to the type we already have or replace/rename the existing types to DDS::*Seq.

Reply:
OK, I think I'm starting to understand what's going on here. The XTypes Annex C IDL defines its own series of `typedef sequence<X> YSeq;` which overlap with, but are not completely the same as, other ones like `OctetSeq` and `StringSeq` in the core spec's IDL.

The comments in DdsDcps.mpc describe the different code generation strategies.  One way to handle this would be to refactor this IDL file and pull out any types that need `opendds_idl` to generate XCDR serialization support.

Another strategy would be to treat these new types like `ByteSeq` as containers that have a C++ mapping but do not have XCDR serializers generated by opendds_idl.  This is the same as `XTypes::Sequence<T>` in `TypeObject.h`.  The only real trick is keeping track of when DHeader is used in XCDR2. Aside from that they're very simple.